internal/testrunner/runners/system/runner.go

```diff
@@ -917,6 +967,9 @@ func (r *runner) prepareScenario(ctx context.Context, config *testConfig, svcInf
 		}
 	}
 	r.deleteTestPolicyHandler = func(ctx context.Context) error {
+		if r.options.RunTestsOnly {
+			return nil
+		}
```

Comment:
Handlers have been updated depending on whether or not they should be running with `--no-provision` flag.